### ISSUE TITLE
scrollbar-position changes:  ctrl+a and then right-click displaces the scroll

### ISSUE THREAD

**Issue by ufateh** [Microsoft/vscode#1448]
open html or any file, big enough so that you can scroll. select all text by ctrl+a and then right-click on it, scrollbar changes its current position to end of the file.

**Comment by DerTolleEmil** [comment]
At least with 0.10.4 I cannot reproduce this. However, the behavior of ctrl+a varies between various editors. All tested on Windows.

VSCode places the cursor at the end of the selection and only scrolls to the cursor again after selecting an item in the context menu.

Notepad, too, places the cursor at the end of the selection but scrolls down immediately. 

Notepad++ places the cursor at the beginning of the selection and and doesn't scroll to the cursor even after selecting copy in the context menu.

Sublime (v3 beta) puts the cursor at the end of the selection and doesn't scroll anywhere either.

Basically out of the four editors I tried every one behaves slightly different. 

I have no idea which one is the correct behavior but VSCode's behavior doesn't seem wrong to me. There is one inconsistency though: When you choose Edit -> Copy the view won't scroll to the cursor but it does if you right click a selection and choose copy.

**Comment by ufateh** [comment]
Another Inconsistency (0.10.5): After selection (ctrl+a), if we use the short-cut key for context menu option (e.g. format code alt+shift+f), view won't scroll, but it scrolls to end if we use click on context menu option.

**Comment by bpasero** [comment]
@alexandrudima I can reproduce this on Mac by right clicking on a file where everything is selected.

**Comment by alexandrudima** [comment]
The contextmenu reveals the editor position right before displaying itself to cover the case that it is invoked via the <kbd>ContextMenu</kbd> key. That's because it needs to figure out a location where to render itself and the best location is the cursor location. Currently, due to some tricks Electron does to render the native context menu, the entire JS VM is paused and the editor paint only occurs after the context menu is closed, thus making it seem that the closing of the context menu caused the jump. I will look into a decent solution.

**Comment by alexandrudima** [comment]
While poking around I've also discovered the following not working case:
* put cursor at the end of the file
* scroll to the top of the file
* press the <kbd>ContextMenu</kbd> key
* the context menu is shown, most likely somewhere off-screen

**Comment by alexandrudima** [comment]
My change makes it that the cursor position gets revealed only if the contextmenu is invoked via a keybinding (in which case it draws itself at the cursor position). So, right clicking will no longer change scroll position whatsoever.
